app/src/main/java/com/example/androidseminar/SignUpActivity.kt

```diff
@@ -1,11 +1,75 @@
 package com.example.androidseminar
 
+import android.app.Activity
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.example.androidseminar.databinding.ActivitySignInBinding
+import com.example.androidseminar.databinding.ActivitySignUpBinding
 
 class SignUpActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySignUpBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_sign_up)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        initButtonClickEvent()
+        Log.d("lifeCycle", "SignUp_onCreate")
+    }
+
+    private fun initButtonClickEvent() {
+        binding.btnSignUp.setOnClickListener {
+
+            val userName = binding.edtName.text
+            val userId = binding.etId.text
+            val userPassword = binding.etPassword.text
+
+            if(userName.isNullOrBlank() || userId.isNullOrBlank() || userPassword.isNullOrBlank()) {
+                Toast.makeText(this@SignUpActivity, "빈 칸이 있는지 확인해주세요", Toast.LENGTH_SHORT).show()
+            }else {
+                //화면 이동
+                val intent = Intent()
+                intent.putExtra("name", binding.edtName.text.toString())
+                intent.putExtra("id", binding.etId.text.toString())
+                intent.putExtra("password", binding.etPassword.text.toString())
```

Comment:
이렇게도 쓸 수 있으니 참고 부탁해요 :)
```suggestion
                intent.putExtra("userName", userName.toString())
                .putExtra("userId", userId.toString())
                .putExtra("userPw", userPw.toString())
```

Reply:
intent를 생략할 수 있구나! 굿